src/core.js

```diff
@@ -217,7 +217,11 @@ jQuery.extend( {
 		// that can be coerced to finite numbers (gh-2662)
 		var type = jQuery.type( obj );
 		return ( type === "number" || type === "string" ) &&
-			( obj - parseFloat( obj ) + 1 ) >= 0;
+
+			// parseFloat NaNs numeric-cast false positives ("")
+			// ...but misinterprets leading-number strings, particularly hex literals ("0x...")
+			// subtraction forces infinities to NaN
+			!isNaN( obj - parseFloat( obj ) );
```

Comment:
Mostly [code golfing for smaller size](https://github.com/jquery/jquery/pull/1352#discussion-diff-6190735). It took awhile before the [`+ 1` was added](https://github.com/stevemao/jquery/commit/e1192af8726bcd25da58aba64b20d89e919230b9), which tipped the scales in favor of the original.

For what it's worth, which wins here between `!isNaN` and `isFinite`?